### PR TITLE
adding sudo rights to travis

### LINKED ISSUE #540: Deploy Every PR on Surge
### Expected results
Automatically generate previews for PR

### Actual results
PR from branches other than gh-pages need to be uploaded somewhere else

### Steps to reproduce the issue
- Add surge.sh integration

### Would you like to work on this
Yes, but need assistance

### PR DESCRIPTION
**Pull request by YashKumarVerma**
- [x] Read and understood [see CONTRIBUTING.md](https://github.com/fossasia/labyrinth/blob/master/CONTRIBUTING.md)
- [x] Included a preview link and screenshot showing after and before the changes.
- [x] Included a description of change below.
- [x] Squashed the commits.

### Changes done in this Pull Request
Now every PR will be deployed on a custom SUB Domain. 

Was proposed since long, finally did it 🔥 
Fixes #540

### CODE REVIEW

File: scripts/deploy-surge.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+if [ “$TRAVIS_PULL_REQUEST” == “false” ]; then
```
Comment (niccokunzmann) [inline scripts/deploy-surge.sh]: You can also test if this is the master branch as not all branches are always deployed.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
-language: node
+sudo: required
+dist: trusty
+language: node_js
 
 script:
   - scripts/check-javascript-syntax-errors.sh
   - scripts/check-html-syntax-errors.sh
+  - bash scripts/deploy-surge.sh
```
Comment (niccokunzmann) [inline .travis.yml]: If you mark this script as executable with 

    chmod +x scripts/deploy-surge.sh

You do not need bash in front of it.

File: scripts/deploy-surge.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+fi
+
+npm i -g surge
+
+export SURGE_LOGIN=test@example.co.in
+# Token of a dummy account
+export SURGE_TOKEN=d1c28a7a75967cc2b4c852cca0d12206
+
+export DEPLOY_DOMAIN=https://pr-${TRAVIS_PULL_REQUEST}-fossasia-labyrinth.surge.sh
+surge —project ./ —domain $DEPLOY_DOMAIN;
```
Comment (niccokunzmann) [inline scripts/deploy-surge.sh]: If you look at the other scripts, you may notice 

    cd "`dirname \"$0\"`"

which makes it so the script can be used from all locations.

**Comment by niccokunzmann** [comment]
@YashKumarVerma I left three comments and the build is failing. What do you think?

**Comment by YashKumarVerma** [comment]
@niccokunzmann 
We need sudo rights in travis to achieve the same. I'm converting this PR to "add sudo rights in travis"

**Comment by YashKumarVerma** [comment]
[ the pr included some changes from a previous commit, that's why COADACY is failing]

Closing this, and opening a new one.

**Pull request closed without merging**